src/gwc/src/main/java/org/geoserver/gwc/config/GeoserverXMLResourceProvider.java

```diff
@@ -39,33 +46,64 @@
     
     private String templateLocation;
 
-    public GeoserverXMLResourceProvider(final String configFileDirectory,
-            final String configFileName,
-            final ResourceStore resourceStore) throws ConfigurationException {
-        
+    public GeoserverXMLResourceProvider(String providedConfigDirectory, String configFileName,
+                                        ResourceStore resourceStore) throws ConfigurationException {
         this.configFileName = configFileName;
-        
-        if(configFileDirectory != null) {
-            // Use the given path
-            if ((new File(configFileDirectory)).isAbsolute()) {
-                
-                log.info("Provided configuration directory as absolute path '" + configFileDirectory + "'");
-                this.configDirectory = Files.asResource(new File(configFileDirectory));
-            } else {
-                log.info("Provided configuration directory in the resource store. ");
-                this.configDirectory = resourceStore.get(configFileDirectory);
-            }
-        } else {
-            this.configDirectory = resourceStore.get(DEFAULT_CONFIGURATION_DIR_NAME);
-        }
-        log.info("Will look for geowebcache.xml in '" + configDirectory + "'");
+        this.configDirectory = inferConfigDirectory(resourceStore, providedConfigDirectory);
+        LOGGER.info(String.format(
+                "Will look for '%s' in directory '%s'.",
+                configFileName, configDirectory.dir().getAbsolutePath()));
     }
     
     public GeoserverXMLResourceProvider(final String configFileName,
             final ResourceStore resourceStore) throws ConfigurationException {
         this(null, configFileName, resourceStore);        
     }
-   
+
+    /**
+     * Helper method that infers the directory that will contain or contains GWC configuration. First we will
+     * check if a specific location was set using property GEOWEBCACHE_CACHE_DIR, then we will check if a
+     * location was provided and then fallback on the default location.
+     */
+    private static Resource inferConfigDirectory(ResourceStore resourceStore, String providedConfigDirectory) {
+        // first let's see if a specific location was defined using GEOWEBCACHE_CONFIG_DIR property
+        String configDirectoryPath = System.getProperty(GEOWEBCACHE_CONFIG_DIR_PROPERTY);
```

Comment:
Something seems odd here; we have a utility method for these kind of settings that respects looking things up as an environmental variable, system property or context parameter.

```
String proxyBase = GeoServerExtensions.getProperty("PROXY_BASE_URL")
```

And the similar logic here - https://github.com/geoserver/geoserver/blob/master/src/platform/src/main/java/org/geoserver/platform/GeoServerResourceLoader.java#L601